src/crimson/osd/osd.cc

```diff
@@ -97,7 +97,7 @@ seastar::future<> OSD::mkfs(uuid_d cluster_fsid)
   }).then([cluster_fsid, this] {
     store->write_meta("ceph_fsid", cluster_fsid.to_string());
     store->write_meta("whoami", std::to_string(whoami));
-    return seastar::now();
+    return store->umount();
```

Comment:
shall we do this in `stop()`?

Reply:
let me try.  Seems OK.

Reply:
I have discussed with Yingxin, msgr shutdown should be executed before engine exit, I am not sure if we need store sharded later, so we'd better keep store->umount out of osd stop function.

Reply:
> msgr shutdown should be executed before engine exit,

agreed. but i am not sure how this is related.

> so we'd better keep `store->umount` out of osd stop function.

probably having `umount()` in `stop()` is not in the scope of this changeset. let's defer making this decision.